dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/visualization/DefaultVisualizationService.java

```diff
@@ -86,4 +92,43 @@ public Visualization getVisualizationNoAcl( String uid )
     {
         return visualizationStore.getByUidNoAcl( uid );
     }
+
+    @Override
+    @Transactional
+    public void removeExpressionDimensionItem( ExpressionDimensionItem expressionDimensionItem )
+    {
+        List<Visualization> visualizations = visualizationStore.getAll()
```

Comment:
This could create a performance issue if we have many Visualizations.
I believe that a better approach would be to implement a concrete VisualizationStore, like `HibernateEventVisualizationStore`. So you can write a specific query to fetch only Visualizations that match the given `expressionDimensionItem.getUid()`.

Reply:
Agreed. Some instances have tens of thousands of saved visualizations.